subsys/usb/class/dfu/Kconfig

```diff
@@ -29,4 +29,10 @@ config USB_DFU_DEFAULT_POLLTIMEOUT
 	help
 	  Default value for bwPollTimeout (in ms)
 
+config USB_DFU_ENABLE_UPLOAD
+	bool "Enable firmware uploading to the host"
+	help
+	  Enabling this option can lead to uploading of decrypted firmware
```

Comment:
Hi,  this description is not very clear

Reply:
Enabling this option allows to download the image by the host. Be aware that the primary image (executable one) is always plain-text despite the image encryption is enabled.

Reply:
Maybe do the `Security Risk: Enabling this option ...` to make it more clear

Reply:
Is is upload / uploading, device uploads the firmware to the host, downloading is _from host to device_.  Clearly described according to the USB DFU spec.

Reply:
@jfischer-no Yes, let use `upload` word:
Security Risk: Enabling this option allows to upload the image to the host. Be aware that the primary image (executable one) is always plain-text despite the image encryption is enabled.

Reply:
What about single slot?

Reply:
Single slot update works with mcuboot serial recovery and allows decryption in place on serial upload. This means that allowing download is risk in application running from single slot if it has been uploaded as encrypted.

Reply:
Unfortunately we do not, yet,  have a method to get information from on mcuboot configuration to resolve this at application runtime. Application also does not know whether it has been uploaded encrypted.

Reply:
> Decrypted firmware is correct term in this case, not plain-text.

OK

Reply:
Updated description